src/common/inject/inject.js

```diff
@@ -158,6 +158,11 @@ Zotero.Inject = new function() {
 					if(translators.length) {
 						me.translators = translators;
 						for(var i=0; i<translators.length; i++) {
+							translators.bestPriority = translators[0].priority;
+							translators.targetURL = document.location.href;
+							translators.targetsTopFrame = document.defaultView == document.defaultView.top;
```

Comment:
This could throw due to SOP. You can just use `isTopWindow` (defined in this file using `try`/`catch`). Also these don't need to be in a loop.